### PR TITLE
add prototype for memrise chats venue

### PR DESCRIPTION
**Pull request by thecadams**

### CODE REVIEW

**Comment by thecadams** [comment]
Closing in favor of @denisdimitrov 's PR:

**Comment by thecadams** [comment]
@denisdimitrov there's a whole boatload of changes here that are not necessary like the bike mode, the gate coordinates. The implementation you've brought over is based around holding the button down, whereas what you want is going to be a "press the button once to move one cell over".

Did something go wrong with your merge? Just trying to understand what's happened.

**Comment by denisdimitrov** [comment]
I was not aware that we want 1 cell movement, I thought the functionality was supposed to be like on the playa and that's what I've been working on until now. My bad. I will try to fix it by the start of the call.

File: src/components/templates/AvatarGrid/AvatarGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+import PARTY_BACKGROUND from "./party-bg.jpg";
+
+type Props = {
+  venueName: string;
+};
+
+const DEFAULT_COLUMNS = 40;
+const DEFAULT_ROWS = 25;
+
+const WIDTH = 2451;
```
Comment (thecadams) [inline src/components/templates/AvatarGrid/AvatarGrid.tsx:23]: Just wondering - how will this make the grid look on mobile? What about on an iPad, or an old laptop? Is fixed width and height the right approach here? Can we use reponsive breakpoints or flex with max-width so the grid cells shrink but the grid scrolls vertically, for example?

File: src/components/molecules/NavBar/NavBar.tsx [hunk 2]
```diff
@@ -29,15 +30,15 @@ import { GiftTicketModal } from "../../organisms/GiftTicketModal/GiftTicketModal
...
+import { VenueTemplate } from "types/VenueTemplate";
 
 interface PropsType {
   redirectionUrl?: string;
 }
 
 const NavBar: React.FunctionComponent<PropsType> = ({ redirectionUrl }) => {
   const { user, profile } = useUser();
-  const venueId = useVenueId();
+  const { venueId } = useParams();
```
Comment (thecadams) [inline src/components/molecules/NavBar/NavBar.tsx:41]: you might find this doesn't work, you should test very carefully. I was seeing venueId as `undefined` in nested routing contexts such as routes in the `CampRouter`, so I suspect you should use the `useVenieId` hook.

File: src/components/templates/AvatarGrid/AvatarGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+
+const DEFAULT_COLUMNS = 40;
+const DEFAULT_ROWS = 25;
+
+const AvatarGrid = ({ venueName }: Props) => {
+  const [selectedUserProfile, setSelectedUserProfile] = useState<
+    WithId<User>
+  >();
+
+  const venueId = venueName;
```
Comment (thecadams) [inline src/components/templates/AvatarGrid/AvatarGrid.tsx:27]: I suspect this is problematic, as we normally convert the venue ID. You can use `const venueId = useVenueId()` instead though.

File: src/components/templates/AvatarGrid/AvatarGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+    return keyPressed;
+  };
+
+  const downPress = useKeyPress("ArrowDown");
+  const upPress = useKeyPress("ArrowUp");
+  const leftPress = useKeyPress("ArrowLeft");
+  const rightPress = useKeyPress("ArrowRight");
+
+  useEffect(() => {
+    const currentPosition = profile?.data?.memrisechats;
```
Comment (thecadams) [inline src/components/templates/AvatarGrid/AvatarGrid.tsx:114]: This is going to affect ALL venues with this template. We need to make it based on venueId.

File: src/components/templates/Playa/PlayaAdmin.tsx [hunk 5]
```diff
@@ -7,17 +7,20 @@ const PlayaAdmin: React.FC = () => {
   const [error, setError] = useState<string | null>();
 
   const playaVenue = useSelector((state) => state.firestore.data.currentVenue);
+  const PLAYA_VENUE_ID = "paddock";
```
Comment (thecadams) [inline src/components/templates/Playa/PlayaAdmin.tsx:10]: Already available in `settings.ts`

File: src/pages/VenuePage/VenuePage.tsx [hunk 6]
```diff
@@ -208,6 +209,12 @@ const VenuePage = () => {
       template = <AudienceRouter />;
       fullscreen = true;
       break;
+    // case VenueTemplate.avatargrid:
```
Comment (thecadams) [inline src/pages/VenuePage/VenuePage.tsx:212]: Should this be commented out? A quick test of a venue should tell you.

**Review by thecadams — COMMENTED** [review]
Looking good with some minor things to look at, looking forward to getting this merged!

**Comment by thecadams** [comment]
Looking good! Going to merge! Thanks to both of you @denisdimitrov and @aivonix !

**Pull request merged**